variable/include/scipp/variable/bucket_model.h

```diff
@@ -17,39 +17,62 @@
 
 namespace scipp::variable {
 
+template <class Indices> class BinModelBase : public VariableConcept {
+public:
+  BinModelBase(const VariableConstView &indices, const Dim dim)
+      : VariableConcept(indices.dims()), m_indices(indices), m_dim(dim) {}
+
+  bool hasVariances() const noexcept override { return false; }
+  void setVariances(Variable &&) override {
+    throw except::VariancesError("This data type cannot have variances.");
+  }
+  VariableConstView bin_indices() const override { return indices(); }
+
+  const auto &indices() const { return m_indices; }
+  auto &indices() { return m_indices; }
+  Dim dim() const noexcept { return m_dim; }
```

Comment:
Rename to something more descriptive. In this class on its own it's not too bad but the derived class has more dimensions. In that context it is not clear what `DataModel<bucket,T>>::dim()` returns.